### DISCUSSION TITLE
Menu Shortcuts

### DISCUSSION THREAD

**Discussion by EdNekebno** [EdNekebno/DGTCentaurMods#32] [Show and tell]
More informational than anything else. Because e-paper is quite slow the menu is quite slow when navigating with the up and down buttons. I therefore coded in shortcut method:

If, when a menu is displayed, the 4th rank is clear on the board then place a piece on that square will make it beep and then removing it will select the matching item. a4 would choose the first one, b4 the second, c4 the third, and so on.

Obviously that's limited to 8 items for now, if we end up getting more we can extend it down to the 3rd rank.

**Comment by wormstein** [comment]
Very cool hack! I noticed too that the menu is quite slow but it's interesting that the original centaur software does better. I don't think it's the epaper delay here but the polling frequency.
The centaur software does the polling like ebery .2 seconds. I think there is a separate thread running that handles this jos only.

**Comment by EdNekebno** [comment]
Polling helps. In eboard.py I use a separate thread for it. But I think the bigger difference is that the original centaur uses partialdisplay updates on the epaper. I could never get that to work for a column for the pointer. I think it may only work on a row basis - which is probably why the centaur draws a box around the selected item. I've been meaning to look. However, this trick works nicely for now.
